client/src/components/Navbar/Navbar.js

```diff
@@ -17,9 +18,28 @@ function Navbar() {
     //When visiting home section
     currenttab = "home";
   }
+  
   const [active, setActive] = useState(currenttab);
   const [navOpen, toggleNav] = useState(0);
 
+  // Handling the case for back button press.
+  window.onpopstate=function()
+  {
+      const currentloc = String(window.location); //Gets current URL
+      let currenttab = currentloc.substring(currentloc.lastIndexOf("/") + 1); //Gets current element from navbar
+
+      if (currenttab[currenttab.length - 1] === "?") {
+        //When search bar is used, preventing bug of nothing being active in navbar
+        currenttab = currenttab.substring(0, currenttab.length - 1); //Removing ? sign when search bar used
+      }
+
+      if (currenttab.length === 0) {
+        //When visiting home section
+        currenttab = "home";
+      }
```

Comment:
marking this as a temporary change because there is code duplication! 
LGTM for now